src/gt4py/eve/traits.py

```diff
@@ -59,7 +59,13 @@ def visit_Node(self, node: concepts.Node, /) -> None:
                         )
                     self.collected_symbols[symbol_name] = node
 
-            if not isinstance(node, SymbolTableTrait):
+            if hasattr(node.annex, "propagated_symbols"):
```

Comment:
Please add a TODO here for me to review and generalize the mechanism to solve name collisions in the `SymbolTableTrait`